src/site/controllers/ticketpurchase.php

```diff
@@ -252,10 +229,10 @@ private function addTicketToDb($memberId, $eventTicketId, $totalCost, $details)
 			);
 			$error_msg = "Oops! There was an error  - DO NOT try again!\r\n";
 			$error_msg .= "Please contact webmaster@swa.co.uk ASAP to resolve this.\r\n";
-			die($error_msg);
+			http_response_code(500);
+			echo json_encode(['error' => "$error_msg"]);
+			die();
 		}
-
 		$this->logAuditFrontend('Member ' . $memberId . ' bought event ticket ' . $eventTicketId);
```

Comment:
I wonder if this should go after getting the intent from Stripe but before trying to insert it into the DB so that there is a log record if adding to the DB fails.